### PR TITLE
[DF] Simplify reading of RDataSource columns 

### PR DESCRIPTION
**Pull request by eguiraud**
Before this commit, RDataSource columns were treated like a special
kind of Defined columns: they were registered in RBookedCustomColumn
and their contents were accessed via RCustomColumn::Get.

This commit removes the logic that was Define'ing ad-hoc columns
corresponding to the RDS columns. Instead, we store the RDS column value
pointers in a dedicated std::map and teach RDSColumnReader to directly
use that. Logic is simpler, we avoid an extra function call and an extra
copy upon data-source value accesses, and we move closer to implementing
column readers specialized for a given RDS implementation.

Sorry for the large commit, unfortunately changing how datasource columns _work_ requires touching a lot of different places and I am not sure it makes sense to split the commit in smaller pieces.

### CODE REVIEW

File: tree/dataframe/src/RLoopManager.cxx [hunk 1]
```diff
@@ -692,3 +692,13 @@ const ColumnNames_t &RLoopManager::GetBranchNames()
    }
    return fValidBranchNames;
 }
+
+bool RLoopManager::HasDSValuePtrs(const std::string &col) const
+{
+   return fDSValuePtrMap.find(col) != fDSValuePtrMap.end();
+}
+
+void RLoopManager::AddDSValuePtrs(const std::string &col, const std::vector<void *> ptrs)
```
Comment (hageboeck) [inline tree/dataframe/src/RLoopManager.cxx:701]: Why is this a vector BTW? Is this for types that consist of multiple columns of primitive types?

**Review by hageboeck — APPROVED** [review]
It looks like it's doing what the commit messages indicate!

File: tree/dataframe/src/RLoopManager.cxx (hunk 1 above)
Comment (eguiraud) [reply tree/dataframe/src/RLoopManager.cxx:701]: it's one vector element per processing slot (~per thread).

**Comment by eguiraud** [comment]
The failure in `datasource_sqlite` should be fixed by #6214 .

**Pull request merged**